src/main/java/org/broadinstitute/hellbender/utils/io/IOUtils.java

```diff
@@ -987,4 +969,42 @@ public static String getGenomicsDBPath(final String path) {
         }
         return genomicsdbPath;
     }
+
+    /**
+     * Schedule a file to be deleted on JVM shutdown.
+     *
+     * @param fileToDelete file or directory to be deleted at JVM shutdown.
+     */
+    public static void deleteOnExit(final Path fileToDelete){
```

Comment:
I added time ago a method in htsjdk which does a similar think ([IOUtils.deleteOnExit](https://github.com/samtools/htsjdk/blob/master/src/main/java/htsjdk/samtools/util/IOUtil.java#L365)), which has a common shutdown hook to delete all at the same time ([DeleteOnExitPathHook](https://github.com/samtools/htsjdk/blob/master/src/main/java/htsjdk/samtools/util/nio/DeleteOnExitPathHook.java)) mimicking the behavior of the `java.io.File#deleteOnExit`.

I would suggest to add a common hook to delete recursively, and change the name of this method. Like that, it would not create too many threads to delete on exit.

Reply:
@magicDGS Yeah, your's is better than this.  I guess I should mimic what you did.

Reply:
I would also rename this method to `deleteRecursivelyOnExit` to be explicit - otherwise, the difference with the htsjdk version is not clear. Also, it would be nice to backport this to htsjdk at some point, and thus it will be easier to migrate afterwards.